src/charm.py

```diff
@@ -309,6 +341,15 @@ def _unit_ips(self) -> List[str]:
         addresses.append(self_address)
         return addresses
 
+    @property
+    def _replica_set_hosts(self):
+        """Fetch current list of hosts in the replica set.
+
+        Returns:
+            A list of hosts addresses (strings).
```

Comment:
You could add type hinting for this value.

Reply:
good eye!!